### ISSUE TITLE
StageWebView Events

### ISSUE THREAD

**Issue by lizardon** [charlesbihis/actionscript-oauth2#3]
It would be nice if the OAuth2 class could fire off events to notify of StageWebView events. In particular a Load URI event and a Finished With StageWebView event. This is necessary because the StageWebView itself does not have a Load URI event. It only has a Change and Changing events. Knowing when a URI begins loading is necessary to implement a loading indicator. Also a Finished with StageWebView event would be nice to have so the app can close the StageWebView as soon as possible. This is particularly the case with installed apps, where the SageWebView will stay open awkwardly after completing the login, while the oauth token is retrieved in the second step.

**Comment by charlesbihis** [comment]
The OAuth2 class takes, as a parameter, an IGrantType object.  Depending on the grant type you are using, this IGrantType object would take in a StageWebView object as a parameter.  So, the actionscript-oauth2 library doesn't actually create any StageWebView objects internally, it simply uses the one that you pass in during construction.  So, you are be able to listen for any StageWebView events for yourself.  Simply attach any listeners before passing it off to the library.

As for adding a Load-URI and Finished-With-StageWebView events, I don't think those are necessary.  As you mention, there are Change and Changing events, which you can use as a Load-URI event (what's the difference?).  As for a Finished-With-StageWebView event, once the library is done with the StageWebView object (i.e. it has retrieved an access token, or failed), it fires off a GetAccessTokenEvent.  This can be used as your Finished-With-StageWebView event.

To address your case with installed apps, where the StageWebView will stay open after completing the login, it sounds like you simply need to close and dispose of the StageWebView object once the GetAccessTokenEvent has fired.  Anyways, I hope this addresses your concerns.  I'm going to close this issue out.  If you still feel like an enhancement is still in order, feel free to re-open or create a new issue.

Charles
